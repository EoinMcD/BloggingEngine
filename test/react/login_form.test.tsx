import * as React from "react";
import LoginForm from "../../app/javascript/components/login_form";
import { expect, test } from "@jest/globals";
import { fireEvent, getByRole, getByTestId, queryByTestId, render, RenderResult, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import "@testing-library/jest-dom";
import $ from "jquery";

const renderForm = (
  path = "/log",
  token = "token"
) : RenderResult => {
  return render(
        <LoginForm path = {path} token = {token}/>
  );
};
describe("Describe the login form", () => {
  afterEach(() => {
    jest.resetModules();
  });
  test("Component renders as expected", () => {
    const { container } = renderForm();
    expect(container).toMatchSnapshot();
  });
  test("Errors array is filled after finding no user", async () => {
    const { container } = renderForm();
    $.ajax = jest.fn().mockImplementation(() => {
      const fakeResponse = {
        errors: "No User Found"
      };
      return Promise.resolve(fakeResponse);
    });
    fireEvent.click(getByRole(container, "button"));
    await waitFor(() => expect(queryByTestId(container, "Error")).toBeTruthy());
  });
  test("Alert opens when path is wrong", () => {
    const { container } = renderForm("/wrongLogPath");
    window.alert = jest.fn();
    fireEvent.click(getByRole(container, "button"));
    expect(window.alert).toBeCalled();
  });
});
