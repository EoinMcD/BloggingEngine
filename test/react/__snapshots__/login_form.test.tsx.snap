// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders as expected 1`] = `
<div>
  <div>
    <form
      class="login-form"
    >
      <h1>
        LOGIN
      </h1>
       
      <br />
      <div
        style="width: 50px; height: 50px;"
      >
        <fieldset>
          <label>
            Email:
          </label>
          <input
            class="form-control"
            name="email"
          />
          <label>
            Password:
          </label>
          <input
            class="form-control"
            name="password"
            type="password"
          />
          <button
            data-testid="loginButton"
            type="submit"
          >
            LOGIN
          </button>
        </fieldset>
      </div>
    </form>
  </div>
</div>
`;

exports[`Describe the login form Component renders as expected 1`] = `
<div>
  <div>
    <form
      class="login-form"
    >
      <h1>
        LOGIN
      </h1>
       
      <br />
      <div
        style="width: 50px; height: 50px;"
      >
        <fieldset>
          <label>
            Email:
          </label>
          <input
            class="form-control"
            name="email"
          />
          <label>
            Password:
          </label>
          <input
            class="form-control"
            name="password"
            type="password"
          />
          <button
            data-testid="loginButton"
            type="submit"
          >
            LOGIN
          </button>
        </fieldset>
      </div>
    </form>
  </div>
</div>
`;
